### DISCUSSION TITLE
Multiprocess

### DISCUSSION THREAD

**Discussion by savasozer** [PrometheusClientNet/Prometheus.Client#209] [General]
Hi all,
I'm trying to use this library in my multi process service. I'll use a shared memory or memory mapped file to write metrics. To do that, I will write new Counter, Gauge and Histogram classess. I wonder should I re-write whole MetricFactory class to use my new metric classess or can I register that new classes to somewhere ?

**Comment by phnx47** [comment]
Hi @savasozer,

It is not clear for me... 
Do you forked and re-write library? It is up to you what you do in fork. If not, show me some examples what you rewrite.
